src/urllib3/connection.py

```diff
@@ -402,6 +413,56 @@ def connect(self):
             or self.assert_fingerprint is not None
         )
 
+    def _connection_requires_tls_in_tls(self):
+        """
+        Indicates if the current connection requires two TLS connections, one to
+        the proxy and one to the destination server.
+        """
+        # Not supported in < py3 or ssl-less environments.
+        if not six.PY3 or not ssl:
```

Comment:
If we're asked by the user for TLS+TLS proxying we shouldn't quietly not provide it. This should be a proxy/connection error.

Reply:
Agreed! I'll throw an error in this case.